package/PartSeg/common_gui/napari_image_view.py

```diff
@@ -0,0 +1,571 @@
+            if not image_info.layers:
+                continue
+            ranges = [
+                (min(a, b), max(c, d), min(e, f))
+                for (a, c, e), (b, d, f) in itertools.zip_longest(
+                    image_info.layers[0].dims.range, ranges, fillvalue=(np.inf, -np.inf, np.inf)
+                )
+            ]
+
+        visible = [ranges[i] for i in self.viewer.dims.displayed]
+        min_shape, max_shape, _ = zip(*visible)
+        size = np.subtract(max_shape, min_shape)
+        return size, min_shape
+
+    @staticmethod
+    def _shift_layer(layer: Layer, translate_2d):
+        translate = [0] * layer.ndim
+        translate[-2:] = translate_2d
+        layer.translate_grid = translate
+
+    def grid_view(self):
+        """Present multiple images in grid view"""
+        n_row = np.ceil(np.sqrt(len(self.image_info))).astype(int)
+        n_row = max(1, n_row)
+        scene_size, _ = self.images_bounds()
+        for image_info, pos in zip(self.image_info.values(), itertools.product(range(n_row), repeat=2)):
+            translate_2d = np.multiply(scene_size[-2:], pos)
+            for layer in image_info.layers:
+                self._shift_layer(layer, translate_2d)
+
+            if image_info.mask is not None:
+                self._shift_layer(image_info.mask, translate_2d)
+
+            if image_info.segmentation is not None:
+                self._shift_layer(image_info.segmentation, translate_2d)
+        self.viewer.reset_view()
+
+    def change_visibility(self, name: str, index: int):
+        for image_info in self.image_info.values():
+            if len(image_info.layers) > index:
+                image_info.layers[index].visible = self.channel_control.channel_visibility[index]
+                if self.channel_control.channel_visibility[index]:
+                    image_info.layers[index].colormap = self.convert_to_vispy_colormap(
+                        self.channel_control.selected_colormaps[index]
+                    )
+                    limits = self.channel_control.get_limits()[index]
+                    limits = image_info.image.get_ranges()[index] if limits is None else limits
+                    image_info.layers[index].contrast_limits = limits
+                    image_info.layers[index].gamma = self.channel_control.get_gamma()[index]
+                    filter_type = self.channel_control.get_filter()[index]
+                    if filter_type != image_info.filter_info[index]:
+                        image_info.layers[index].data = self.calculate_filter(
+                            image_info.image.get_channel(index), filter_type
+                        )
+                        image_info.filter_info[index] = filter_type
+
+    def reset_image_size(self):
+        self.viewer.reset_view()
+
+    def set_theme(self, theme: str):
+        self.viewer.theme = theme
+
+
+class NapariQtViewer(QtViewer):
+    def dragEnterEvent(self, event):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/Czaki/PartSeg/pullRequest?prid=5772320)